crates/subspace-gateway/src/node_client.rs

```diff
@@ -0,0 +1,61 @@
+//! Node client implementation that connects to node via RPC (WebSockets)
+
+use async_trait::async_trait;
+use jsonrpsee::core::client::{ClientT, Error as JsonError};
+use jsonrpsee::rpc_params;
+use jsonrpsee::ws_client::{WsClient, WsClientBuilder};
+use std::fmt;
+use std::sync::Arc;
+use subspace_core_primitives::segments::{SegmentHeader, SegmentIndex};
+use subspace_rpc_primitives::FarmerAppInfo;
+
+/// Node client implementation that connects to node via RPC (WebSockets).
+///
+/// This implementation is supposed to be used on local network and not via public Internet due to
+/// sensitive contents.
+#[derive(Debug, Clone)]
+pub struct RpcNodeClient {
+    client: Arc<WsClient>,
+}
+
+impl RpcNodeClient {
+    /// Create a new instance of [`NodeClient`].
+    pub async fn new(url: &str) -> Result<Self, JsonError> {
+        let client = Arc::new(WsClientBuilder::default().build(url).await?);
+        Ok(Self { client })
+    }
+}
```

Comment:
This is fine for now, but it is a hack. I think we'll need a separate RPC for these purposes.